crml/nft/src/lib.rs

```diff
@@ -0,0 +1,651 @@
+		/// Too many attributes in the provided schema or data
+		SchemaMaxAttributes,
+		/// Provided attributes do not match the collection schema
+		SchemaMismatch,
+		/// Provided attribute is not in the collection schema
+		UnknownAttribute,
+		/// The provided attributes or schema cannot be empty
+		SchemaEmpty,
+		/// The schema contains an invalid type
+		SchemaInvalid,
+		/// The schema contains a duplicate attribute name
+		SchenmaDuplicateAttribute,
+		/// Given attirbute value is larger than the configured max.
+		MaxAttributeLength,
+		/// origin does not have permission for the operation
+		NoPermission,
+		/// The NFT collection does not exist
+		NoCollection,
+		/// The NFT does not exist
+		NoToken,
+		/// The NFT is not listed for a direct sale
+		NotForDirectSale,
+		/// The NFT is not listed for auction sale
+		NotForAuction,
+		/// Cannot operate on a listed NFT
+		TokenListingProtection,
+		/// Internal error during payment
+		InternalPayment,
+		/// Total royalties would exceed 100% of sale
+		RoyaltiesOvercommitment,
+		/// Auction bid was lower than reserve or current highest bid
+		BidTooLow
+	}
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Nft {
+		/// Map from collection to owner address
+		pub CollectionOwner get(fn collection_owner): map hasher(blake2_128_concat) CollectionId => Option<T::AccountId>;
+		/// Map from collection to its schema definition
+		pub CollectionSchema get(fn collection_schema): map hasher(blake2_128_concat) CollectionId => Option<NFTSchema>;
+		/// Map from collection to it's defacto royalty scheme
+		pub CollectionRoyalties get(fn collection_royalties): map hasher(blake2_128_concat) CollectionId => Option<RoyaltiesSchedule<T::AccountId>>;
+		/// Map from a token to it's royalty scheme
+		pub TokenRoyalties get(fn token_royalties): double_map hasher(blake2_128_concat) CollectionId, hasher(twox_64_concat) T::TokenId => Option<RoyaltiesSchedule<T::AccountId>>;
+		/// Map from (collection, token) to it's attributes (as defined by schema)
+		pub Tokens get(fn tokens): double_map hasher(blake2_128_concat) CollectionId, hasher(twox_64_concat) T::TokenId => Vec<NFTAttributeValue>;
+		/// The next available token Id for an NFT collection
+		pub NextTokenId get(fn next_token_id): map hasher(twox_64_concat) CollectionId => T::TokenId;
+		/// The total amount of an NFT collection in existence
+		/// Map from (collection, token) to it's owner
+		pub TokenOwner get(fn token_owner): double_map hasher(blake2_128_concat) CollectionId, hasher(blake2_128_concat) T::TokenId => T::AccountId;
+		/// Map from (collection, account) to the account owned tokens of that collection
+		pub CollectedTokens get(fn collected_tokens): double_map hasher(blake2_128_concat) CollectionId, hasher(blake2_128_concat) T::AccountId => Vec<T::TokenId>;
+		/// The total amount of an NFT collection in existence
+		pub TokenIssuance get(fn token_issuance): map hasher(blake2_128_concat) CollectionId => T::TokenId;
```

Comment:
I'd say for this purpose adding an extrinsic that would count `Tokens` is a better idea.